### PR TITLE
Issues/1733 reset entity view id holder on success and failure

### LINKED ISSUE #1733: Possible entity view id leakage in Spring WebMVC integration
### Description
`com.blazebit.persistence.spring.data.webmvc.impl.json.EntityViewIdHandlerInterceptor` is not guaranteed to reset the ThreadLocal in  `com.blazebit.persistence.spring.data.webmvc.impl.json.EntityViewIdValueHolder` because the wrong handler method is used. The current implementation uses `postHandle` which is only invoked on handler success. Instead, `afterCompletion` should be used.

### PR DESCRIPTION
**Pull request by Mobe91**
Fixes #1733

### CODE REVIEW

File: integration/spring-data/testsuite/webmvc/pom.xml [hunk 1]
```diff
@@ -125,6 +125,56 @@
         </dependency>
     </dependencies>
 
+    <build>
```
Comment (beikov) [inline integration/spring-data/testsuite/webmvc/pom.xml:128]: The `<id>hibernate-apt</id>` profile already includes the annotation processor for testing. Is that not enough?

Comment (Mobe91) [reply integration/spring-data/testsuite/webmvc/pom.xml:128]: This is about the entity-view-processor, not about the Hibernate static metamodel.

Comment (beikov) [reply integration/spring-data/testsuite/webmvc/pom.xml:128]: That profile also generate the entity view static metamodel classes.

**Review by beikov — APPROVED** [review]

**Comment by beikov** [comment]
Ahh, seems I didn't see you are using the builder in `DocumentControllerTest`. In that case, it might be better to simply make sure the hibernate-apt profile still works properly even if you declare the bsc-plugin in the main build section.

**Pull request merged**